lib/browser/extensions/checksum.js

```diff
@@ -0,0 +1,39 @@
+class Checksum {
+  static supportedAlogrithms = ['SHA-1', 'SHA-256', 'SHA-384', 'SHA-512'];
+
+  constructor (algo = 'SHA-256') {
+    this.algo = this._resolveAlgo(algo)
+  }
+
+  /**
+   * Resolves algorithm to one of the inbuilt types
+   * @param {String} algo contains user provided checksumAlgo option
+   */
+  _resolveAlgo = (algo) => {
+    const resolvedAlgo = Checksum.supportedAlgorithms.find(supportedAlgo => supportedAlgo === algo.toUpperCase())
```

Comment:
That logic could be simplified using `Array#includes`. All in all, the _resolveAlgo method seems a bit unnecessary, as this could all be done in the constructor directly.